src/ranges.jl

```diff
@@ -214,10 +222,17 @@ unsafe_length_one_to(::StaticInt{L}) where {L} = L
     elseif step > 0
         return Base.checked_add(Int(div(Base.checked_sub(stop, start), step)), 1)
     else
-        return Base.checked_add(Int(div(Base.checked_sub(rtart, stop), -step)), 1)
+        return Base.checked_add(Int(div(Base.checked_sub(start, stop), -step)), 1)
     end
 end
 
+@propagate_inbounds function Base.getindex(r::OptionallyStaticUnitRange, s::AbstractUnitRange{<:Integer})
+    @boundscheck checkbounds(r, s)
+    f = static_first(r)
+    f = f - one(f)
```

Comment:
I'd rename `f` in case of type instabilities. This shows up in `@code_warntype` for me.